scaleway/data_source_vpc_private_network.go

```diff
@@ -71,5 +84,47 @@ func dataSourceScalewayVPCPrivateNetworkRead(ctx context.Context, d *schema.Reso
 	zonedID := datasourceNewZonedID(privateNetworkID, zone)
 	d.SetId(zonedID)
 	_ = d.Set("private_network_id", zonedID)
-	return resourceScalewayVPCPrivateNetworkRead(ctx, d, meta)
+	return resourceScalewayVPCPrivateNetworkZonalRead(ctx, d, meta)
+}
+
+func dataSourceScalewayVPCPrivateNetworkRegionalRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vpcAPI, region, err := vpcAPIWithRegion(d, meta)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	privateNetworkID, ok := d.GetOk("private_network_id")
+	if !ok {
+		res, err := vpcAPI.ListPrivateNetworks(
+			&v2.ListPrivateNetworksRequest{
+				Name:   expandStringPtr(d.Get("name").(string)),
+				Region: region,
+			}, scw.WithContext(ctx))
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		if res.TotalCount == 0 {
+			return diag.FromErr(
+				fmt.Errorf(
+					"no private network found with the name %s",
+					d.Get("name"),
+				),
+			)
+		}
+		if res.TotalCount > 1 {
+			return diag.FromErr(
+				fmt.Errorf(
+					"%d private networks found with the name %s",
+					res.TotalCount,
+					d.Get("name"),
+				),
+			)
+		}
+		privateNetworkID = res.PrivateNetworks[0].ID
+	}
+
+	regionalID := datasourceNewRegionalizedID(privateNetworkID, region)
+	d.SetId(regionalID)
+	_ = d.Set("private_network_id", regionalID)
+	return resourceScalewayVPCPrivateNetworkRegionalRead(ctx, d, meta)
```

Comment:
Data source does not fail if given `private_network_id` does not exist

```suggestion
	diags := resourceScalewayVPCPrivateNetworkRegionalRead(ctx, d, meta)
	if diags != nil {
		        return append(diags, diag.Errorf("failed to read private network state")...)
	}
	
	if d.Id() == nil {
		return diag.Errorf("private network (%s) not found", regionalID)
	}
	
	return nil
```